### PR TITLE
Remove MCI, DX, and DIO CD-ROM interfaces

### PR DESCRIPTION
**Pull request by dreamer**
Also, remove all traces of these interfaces from user manual (README
file) and man page (docs/dosbox.1).

MCI (Media Control Interface) was a primary Windows interface. Code
comments and documentation claimed, that it's only for "NT, 2000, XP"
but the code was enabled for Windows 2000 or later (version > 4.0).

DX (Digital audio eXtraction (?)) could be forced on any Windows with
autodetection working only on Vista and Windows 7 (code was disabled for
Windows 8 or later - probably unintentionally).

DIO used DeviceIoControl interface and Windows-specific ioctl calls.

All 3 interfaces depend on SDL and SDL_cdrom functionality to work.
SDL_cdrom 1.2 implementation uses MCI on Windows to provide the same
functionality.

----
Continuation of #29 work; these interfaces are rather different than ASPI removed in #81 - these ones are actually usable, but they duplicate (and depend on) on SDL_cdrom functionality anyway.

If we'll ever want to bring back physical CD-ROM support, then reverting this commit might be the first step towards having non-SDL_cdrom based implementation. But removing it from the code will enable some code refactorizations and cleanup in other CD-ROM interface classes and MSCDEX area.

Please double check that I cleaned it up thoroughly - also in the documentation (hopefully everything in manual still makes sense after I removed references to these interfaces). I have no idea why they were mentioned in man page for Unix systems…

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -134,23 +134,14 @@ CD-ROM: My CD-ROM doesn't work.
...
-      To enable SDL-support (does not include low-level CD access!):
-        - mount d f:\ -t cdrom -usecd 0 -noioctl
-      To enable ioctl access using digital audio extraction for CD audio
-      (Windows-only, useful for Vista):
-        - mount d f:\ -t cdrom -ioctl_dx
-      To enable ioctl access using MCI for CD audio (Windows-only):
-        - mount d f:\ -t cdrom -ioctl_mci
-      To force ioctl-only access (Windows-only):
-        - mount d f:\ -t cdrom -ioctl_dio
+      - mount d f:\ -t cdrom -usecd 0 -noioctl
```
Comment (krcroft) [inline README]: The only remaining mount option is to try via SDL, so suggest rewording it a bit - 
---
You can also append `-usecd 0 -noioctl` to access the CDROM via SDL, if the above doesn't work.
For example:
- `mount d f:\ -t cdrom -usecd 0 -noioctl`
- `mount d /media/cdrom -t cdrom -usecd 0 -noioctl`

File: src/platform/visualc/ntddcdrm.h [hunk 2]
```diff
@@ -1,320 +0,0 @@
-/*
- * ntddcdrm.h
- *
- * CDROM IOCTL interface.
- *
- * This file is part of the w32api package.
- *
- * Contributors:
- *   Created by Casper S. Hornstrup <chorns@users.sourceforge.net>
- *
```
Comment (krcroft) [inline src/platform/visualc/ntddcdrm.h]: Interesting.. found that this is now part of MinGW, and more revised versions are out there.  I wonder if this was on anyone's radar to keep it up to date; or if they were aware of it, but choose not to?   In anycase, glad all these OS-specific hooks are being tossed!

More general comment:
A document that lists externally-maintained files like this (same w/ the single-header audio decoders), would be handy for future devs to keep them up-to-date with upstream.

File: docs/dosbox.1 [hunk 3]
```diff
@@ -162,7 +161,7 @@ Sets the amount of free space available on a drive in MB's. This is a more
 .RI "Sets the name of the drive to " drivelabel ". Needed on some" 
 systems if the cd label isn't read correctly. Useful when a 
 program can't find its cdrom. If you don't specify a label and no
-.RB "lowlevel support is selected (" "\-usecd #" " and/or " "\-ioctl" "):"
+.RB "lowlevel support is selected (" "\-usecd #" "):"
```
Comment (krcroft) [inline docs/dosbox.1]: I think `-ioctl` is still valid, because that's using the SDL interface (if I understand correctly?)

File: README [hunk 4]
```diff
@@ -816,7 +793,7 @@ MIXER
   channel
      Can be one of the following: MASTER, DISNEY, SPKR, GUS, SB, FM [, CDAUDIO].
      CDAUDIO is only available if a CD-ROM interface with volume control is
-     enabled (CD image, ioctl_dx).
+     enabled (CD image).
 
   left:right
      The volume levels in percentages. If you put a D in front it will be
```
Comment (krcroft) [inline README]: Also see paragraph at line 1452:
```
  - if the game requires a CD-ROM be sure you used "-t cdrom" when
    mounting and try different additional parameters (the ioctl,
    usecd and label switches, see the appropriate section)
```
Suggest replacing with something like:
```
  - if the game requires a CD-ROM, then be sure you used "-t cdrom" when
    mounting, and refer to the above section "CD-ROM: My CD-ROM doesn't work."
    for more help.
```

File: README [hunk 5]
```diff
@@ -134,23 +134,14 @@ CD-ROM: My CD-ROM doesn't work.
```
Comment (krcroft) [inline README]: (minor fix at  line 49): `If you have your game on a cdrom you may try ...`
Rename `cdrom` to `CD-ROM` to match with consistent naming in the documentation.

File: docs/dosbox.1 [hunk 6]
```diff
@@ -173,9 +172,6 @@ If you do specify a label this label will be kept as long as the drive
 is mounted. It will not be updated !!
 .RE
 .TP
-.B \-ioctl   
-Forces to use ioctl commands.
-.TP
 .BI \-usecd " number"
 Forces to use SDL cdrom support for drive number.
```
Comment (krcroft) [inline docs/dosbox.1]: Interesting that `-noioctl` isn't mentioned in this man-page.
I think adding it out-of-scope from the PR, and our trajectory toward dropping SDL CD-ROM support also doesn't warrant us bothering to keep track of this for future additions)

File: src/dos/cdrom.h [hunk 7]
```diff
@@ -274,102 +271,6 @@ class CDROM_Interface_Image : public CDROM_Interface
 	Bit8u               subUnit;
 };
 
-#if defined (WIN32)	/* Win 32 */
-
```
Comment (krcroft) [inline src/dos/cdrom.h]: Wow; cdrom.h is now very tidy.

File: src/dos/dos_programs.cpp [hunk 8]
```diff
@@ -327,31 +327,8 @@ class MOUNT : public Program {
...
-					if ((osi.dwPlatformId==VER_PLATFORM_WIN32_NT) && (osi.dwMajorVersion>5)) {
-						// Vista/above
-						MSCDEX_SetCDInterface(CDROM_USE_IOCTL_DX, num);
-					} else {
-						MSCDEX_SetCDInterface(CDROM_USE_IOCTL_DIO, num);
-					}
-#else
-					MSCDEX_SetCDInterface(CDROM_USE_IOCTL_DIO, num);
-#endif
 				}
```
Comment (krcroft) [inline src/dos/dos_programs.cpp]: Previously by default, `mount d f:\ -t cdrom` would be an `-ioctl` mount via IOCTL_DX or DIO, which would give them an MSCDEX interface.

With this gone now, SDL is the only remaining option to get an MSCDEX interface; so suggest we make `MSCDEX_SetCDInterface(CDROM_USE_SDL, num);` the default here if nothing is specified. 
If `-ioctl` is specified, then maybe exit with an error?

File: src/dos/dos_programs.cpp [hunk 9]
```diff
@@ -327,31 +327,8 @@ class MOUNT : public Program {
...
-				if (cmd->FindExist("-ioctl_dio", false)) {
-					MSCDEX_SetCDInterface(CDROM_USE_IOCTL_DIO, num);
-				} else if (cmd->FindExist("-ioctl_dx",false)) {
-					MSCDEX_SetCDInterface(CDROM_USE_IOCTL_DX, num);
-#if defined (WIN32)
-				} else if (cmd->FindExist("-ioctl_mci",false)) {
-					MSCDEX_SetCDInterface(CDROM_USE_IOCTL_MCI, num);
-#endif
-				} else if (cmd->FindExist("-noioctl",false)) {
+				if (cmd->FindExist("-noioctl",false)) {
```
Comment (krcroft) [inline src/dos/dos_programs.cpp]: Previously under Windows a default `mount d f:\ -t cdrom` would give them an `-ioctl`-type mount with an MSCDEX interface.
With `-ioctl` now gone under Windows, what happens to `mount d f:\ -t cdrom`?  I think we have two options:
1. Quit with an error and inform the user that implied `-ioctl` mounts are not available, and they must specify `-noioctl` from now on.
1. Switch and make `-noioctl` the default, so they still get an MSCDEX interface via MSCDEX_SetCDInterface.

File: src/platform/visualc/ntddcdrm.h (hunk 2 above)
Comment (dreamer) [reply src/platform/visualc/ntddcdrm.h]: Hmm, yeah - so far (aside from `decoders`) I knew about files from MAME, FreeDOS and gui_tk (which seems unused); simple grep inside src reveals few more files, but there are some false positives as well:

```
$ git grep Copyright | grep -v DOSBox
dos/dos_codepages.h: *  Copyright (C)  Henrique Peron
dos/dos_codepages.h:   Copyright (C) 1993-1996 by kostis@acm.org (Kosta Kostis) */
dos/dos_codepages.h:   Copyright (c) 1996-2002 Markus Oberhumer & Laszlo Molnar
dos/dos_keyboard_layout_data.h: *  Copyright (C)  Henrique Peron
dos/dos_keyboard_layout_data.h:   Copyright (C) 2004 by Aitor SANTAMARIA_MERINO */
dos/dos_mscdex.cpp:		case 0x1502:	/* Get Copyright filename */
hardware/mame/fmopl.cpp:** Copyright Jarek Burczynski (bujar at mame dot net)
hardware/mame/fmopl.cpp:** Copyright Tatsuyuki Satoh , MultiArcadeMachineEmulator development
hardware/mame/ymf262.cpp:** Copyright Jarek Burczynski
hardware/opl.cpp: * Copyright (C) 1998-2001 Ken Silverman
hardware/opl.h: * Copyright (C) 1998-2001 Ken Silverman
hardware/sblaster.cpp:	case 0xe3:	/* DSP Copyright */
libs/decoders/SDL_sound.c: * Copyright (C) 2001  Ryan C. Gordon.
libs/decoders/SDL_sound.h: * Copyright (C) 2001  Ryan C. Gordon.
libs/decoders/SDL_sound_internal.h: * Copyright (C) 2001  Ryan C. Gordon.
libs/decoders/docs/LICENSE.txt: Copyright (C) 1991, 1999 Free Software Foundation, Inc.
libs/decoders/docs/LICENSE.txt:    Copyright (C) <year>  <name of author>
libs/decoders/dr_flac.h:Copyright 2018 David Reid
libs/decoders/dr_mp3.h:Copyright 2018 David Reid
libs/decoders/dr_wav.h:Copyright 2018 David Reid
libs/decoders/stb.h:Copyright (c) 2017 Sean Barrett
libs/decoders/stb_vorbis.h:Copyright (c) 2017 Sean Barrett
libs/decoders/xxh3.h:   Copyright (C) 2019-present, Yann Collet.
libs/decoders/xxhash.c:*  Copyright (C) 2012-2016, Yann Collet
libs/decoders/xxhash.h:   Copyright (C) 2012-2016, Yann Collet.
libs/gui_tk/gui_tk.cpp: *  Copyright (C) 2005-2007 Jörg Walter
libs/gui_tk/gui_tk.h: * Copyright (C) 2005-2007 Jörg Walter
libs/zmbv/drvproc.cpp:// Copyright (c) 1993 Microsoft Corporation.
libs/zmbv/zmbv_vfw.cpp:// Copyright (c) 1993 Microsoft Corporation.
```

File: src/dos/dos_programs.cpp (hunk 9 above)
Comment (dreamer) [reply src/dos/dos_programs.cpp]: Huh, good point; when making this change few days ago I was *convinced*, that all code paths route back to SDL as a fallback in the end, but I'm not 100% sure any more - I'll double check.

If it'll turn out to be true and -ioctl/-noioctl is pointless now on Windows (same way as it always was pointless on macOS), then I'll just print some user warnings.

Overall, this will be a bigger deal when removing last remains of physical support - I marked some places in code, that will need to trigger warnings/errors to the users trying to mount

Comment (dreamer) [reply src/dos/dos_programs.cpp]: OK, I double checked - when `-t cdrom`  is passed, then no matter if `-ioctl`, `-noioctl`, or nothing at all is passed - each time SDL CD-ROM interface will be selected (except via slightly different codepaths). If destination is not a real cdrom, then it will fallback to mounting as directory (as if `-t cdrom` was not passed at all) - I don't like this behaviour, but perhaps it's the best for backwards-compatibility.

Also, on Linux parameters `-ioctl`/`-noioctl` do not make sense, as both fall into ioctl interface inheriting from SDL interface.

Added a warning to notify user, that `-ioctl` does not work any more.

File: README (hunk 1 above)
Comment (dreamer) [reply README]: Under Linux `-ioctl` and `-noioctl` work the same - both trigger "Ioctl" interface, which in case of Linux inherits from SDL interface (and not the pure abstract class). The only difference is, that when `-usecd` works only when `-noioctl` is passed (which is a bug, I presume).

Temporarily, this line works also on non-Linux systems, so... I think I'll leave it as it is - it will be removed very soon anyway.

File: src/dos/dos_programs.cpp [hunk 10]
```diff
@@ -327,7 +327,10 @@ class MOUNT : public Program {
 				int num = -1;
 				cmd->FindInt("-usecd",num,true);
 				int error = 0;
-				if (cmd->FindExist("-noioctl",false)) {
+				if (cmd->FindExist("-ioctl", false)) {
+					WriteOut(MSG_Get("MSCDEX_WARNING_IOCTL"));
+				}
+				if (cmd->FindExist("-noioctl", false)) {
 					MSCDEX_SetCDInterface(CDROM_USE_SDL, num);
 				}
```
Comment (krcroft) [inline src/dos/dos_programs.cpp]: Previous code fell through to the else { } block if neither `-ioctl` or `-noioctl` were explicitly specified.  In the else { } block they got one of the `ioctl` MSCDEX interfaces. 

In this new code, there isn't a default, so they fall through to no MSCDEX interface. 
Looking ahead to SDL2, all of this is headed out the door anyway, so I realize I'm probably sweating too much on this; chances are this is all moot in a couple days anyway :-)
